trussworks/react-uswds
# feat: Upgrade to React 17

# Summary

Upgrade to [React 17](https://reactjs.org/blog/2020/08/10/react-v17-rc.html) (Raised as an issue previously, but accelerated as it proves to be a blocker for implementing `focusout` effects for the DatePicker component)

A (happy) consequence of the upgrade is the need to also upgrade storybook to a React 17 compatible version (latest stable at the moment, v6.1.5), so you'll see that included in this PR.

## Related Issues or PRs

follows #842 
closes #805 
unblocks https://github.com/trussworks/react-uswds/pull/803

## How To Test
Spin up a CRA or verify an existing app depending on this library is still able to use the repo on React 16.

Use [Github URL](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#github-urls) dependency as follows:
`"@trussworks/react-uswds": "https://github.com/trussworks/react-uswds#bl-react-17-805"`

### Screenshot from testing via a create react app:
![image](https://user-images.githubusercontent.com/15805554/106793794-ea8f1200-6625-11eb-94b0-93c0dc5a9965.png)
